src/core/qgsrendercontext.h

```diff
@@ -152,6 +156,17 @@ class CORE_EXPORT QgsRenderContext
     /** Sets pointer to original (unsegmentized) geometry*/
     void setGeometry( const QgsAbstractGeometryV2* geometry ) { mGeometry = geometry; }
 
+    /** Set a filter feature provider used to filter the features
+     * @param ffp the filter feature provider
+     * @note not available in Python bindings
+     */
+    void setFeatureFilterProvider( const QgsFeatureFilterProvider* ffp ) { mFeatureFilterProvider = ffp; }
```

Comment:
this should probably store a clone of ffp - the current approach looks risky to me